### PR TITLE
Refine validate job status script

### PR DESCRIPTION
**Pull request by QiJune**
Please review #1481 first.

This PR does two things:

- support flexible configuration of ps num and worker num
- rewirte validate job status script in python. Bash shell is too hard:)

### CODE REVIEW

**Review by terrytangyuan — CHANGES_REQUESTED** [review]
If we are switching to Python, we should use k8s Python API directly instead of using `kubectl`. Otherwise I am not sure whether the switch is useful since the original bash script was pretty straightforward.

File: scripts/setup_k8s_env.sh [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #!/bin/bash
 
+sudo pip3 install kubernetes
```
Comment (terrytangyuan) [inline scripts/setup_k8s_env.sh:4]: Add a note on where this is used

File: .travis.yml [hunk 2]
```diff
@@ -1,11 +1,12 @@
 language: minimal
 
-dist: xenial
+dist: bionic
```
Comment (terrytangyuan) [inline .travis.yml:3]: Why is this change needed?

Comment (QiJune) [reply .travis.yml:3]: Ubuntu 16.04 LTS reaches End Of Life (EOL). Please refer to https://ubuntustudio.org/2019/04/ubuntu-studio-16-04-lts-reaches-end-of-life-eol/.

I change it to ubuntu 18.04.

File: scripts/setup_k8s_env.sh (hunk 1 above)
Comment (QiJune) [reply scripts/setup_k8s_env.sh:4]: Done

File: scripts/validate_job_status.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+            print(
+                "ElasticDL job succeeded"
+                "(master pod keeps running for TensorBoard service)."
+            )
+            client.delete_pod(master_pod_name)
+            return
+        elif (
+            check_failed(ps_pod_phases)
+            or check_failed(worker_pod_phases)
+            or check_failed([master_pod_phase])
```
Comment (terrytangyuan) [inline scripts/validate_job_status.py:110]: The conditions only check whether the statuses failed but should also exit non-zero if anything of them failed.

Comment (terrytangyuan) [reply scripts/validate_job_status.py:110]: Have you validated the failure cases to see if Travis fails?

Comment (QiJune) [reply scripts/validate_job_status.py:110]: Done.
Please refer to #1493 , I modify the code in PS to make PS pod crash, https://github.com/sql-machine-learning/elasticdl/pull/1493/files#diff-afbc90b2c61dee80a12b74ea4e2a6c91 

The ci status is failed. Please refer to https://travis-ci.org/sql-machine-learning/elasticdl/jobs/614841060?utm_medium=notification&utm_source=github_status

File: scripts/client_test.sh [hunk 4]
```diff
@@ -10,16 +12,16 @@ if [[ "$JOB_TYPE" == "train" ]]; then
       --training_data=/data/frappe/train \
       --validation_data=/data/frappe/test \
       --num_epochs=1 \
-      --master_resource_request="cpu=400m,memory=1024Mi" \
+      --master_resource_request="cpu=0.2,memory=1024Mi" \
       --master_resource_limit="cpu=1,memory=2048Mi" \
-      --worker_resource_request="cpu=400m,memory=2048Mi" \
+      --worker_resource_request="cpu=0.4,memory=2048Mi" \
```
Comment (workingloong) [inline scripts/client_test.sh:17]: What is the difference between `400m` and `0.4`?

**Review by terrytangyuan — APPROVED** [review]

File: scripts/client_test.sh (hunk 4 above)
Comment (QiJune) [reply scripts/client_test.sh:17]: Please refer to https://kubernetes.io/zh/docs/concepts/configuration/manage-compute-resources-container/#cpu-%E7%9A%84%E5%90%AB%E4%B9%89

表达式 0.1 等价于表达式 100m，可以看作 “100 millicpu”
